### ISSUE TITLE
LibSass supports cross-media @extend

### ISSUE THREAD

**Issue by HugoGiraudel** [sass/libsass#712]
I feel kind of dirty reporting this, because this is not actually a bug per se... It turns out LibSass does support something Ruby Sass doesn't. Moreover, this feature has been asked for years in Sass, so I'm quite confused about the way to go here... Should this be removed? I'll leave you the only judge of this.

Test:

```scss
.selector {
  content: selector;
}

@media print {
  .other-selector {
    @extend .selector;
  }
}
```

Expect:

```css
You may not @extend an outer selector from within @media.
You may only @extend selectors within the same directive.
From "@extend .selector" on line 7.
```

Result:

```css
.selector,
.other-selector {
    content: selector;
}
```

Ref: http://sass-compatibility.github.io/#cross_media_extend.

**Comment by HugoGiraudel** [comment]
Maintainers, what is the current status on this?

**Comment by chriseppstein** [comment]
This is a bug in libsass. Sass does have a feature planned to support cross media extends but it does not have the output you see here. The correct output requires Sass to keep the `@media` context when the extend is performed. The feature enhancement is being tracked here: https://github.com/sass/sass/issues/1050

**Comment by chriseppstein** [comment]
@HugoGiraudel The information at http://sass-compatibility.github.io/#cross_media_extend is not accurate. LibSass doesn't "get this right". It gets it very wrong just like sass got it very wrong back when we allowed it.

**Comment by HugoGiraudel** [comment]
Sass Compatibility has no "get it right" or "get it wrong". It has support or no support.

**Comment by chriseppstein** [comment]
The presentation implies there is a feature that libsass has and other implementations don't. It's impossible for Ruby sass 3.4 anything other than 100% because it is the reference implementation. 

Hunt & pecked on my iPhone... Sorry if it's brief!

**Comment by HugoGiraudel** [comment]
I could add a note about how LibSass *gets it wrong* in the description if you will but I won't be able to do much more. Sass Compatibility intends to list inconsistencies between Sass engines. It is an inconsistency, no matter if both are not supporting what you deem is correct.

**Comment by chriseppstein** [comment]
@HugoGiraudel Then the sense of the discrepency is wrong. The feature should be "forbids @extend in @media" and Sass should have it checked and libsass should not have it checked.

**Comment by chriseppstein** [comment]
@HugoGiraudel Anyways, I really like the presentation of your compatibility guide. It would be awesome if we could data-drive it from sass-spec. are you open to working with us on that?

**Comment by HugoGiraudel** [comment]
>  The feature should be "forbids @extend in @media" and Sass should have it checked and libsass should not have it checked.

That seems reasonable.

> we could data-drive it from sass-spec

It already is, except for the few tests that do not exist on sass-spec. I have been working with @xzyfer on this for several weeks now.

**Comment by chriseppstein** [comment]
> It already is

Excellent. Where is the mapping of tests to features?

**Comment by HugoGiraudel** [comment]
Hardly the place for it I suppose. Where should I reach you Chris?

**Comment by matthewpavkov** [comment]
So, libsass does not support @extend within @media, and should not be used, correct?

**Comment by HugoGiraudel** [comment]
LibSass does support a buggy version of it, hence should not be used.

**Comment by xzyfer** [comment]
@HugoGiraudel could you please create a spec representing the correct behaviour?

**Comment by HugoGiraudel** [comment]
There: https://github.com/sass/sass-spec/pulls.

**Comment by xzyfer** [comment]
@mgreter have you confirmed the (unmerged) spec https://github.com/sass/sass-spec/pull/261 passes?

**Comment by xzyfer** [comment]
Nevermind, that spec requires capturing errors.

**Comment by saper** [comment]
I can confirm that for the https://github.com/sass/sass-spec/pull/261 testcase libsass still works, i.e. produces output

````css
.foo, .bar {
  content: 'foo'; }
````

without reporting an error.

**Comment by xzyfer** [comment]
Looks like a regression in 3.3. This was fixed in 3.2.0.

```
Error: You may not @extend an outer selector from within @media.
       You may only @extend selectors within the same directive.
       From "@extend .foo" on line 35 of test.scss
        on line 29 of test.scss
```

**Comment by xzyfer** [comment]
Tracking the regression in https://github.com/sass/libsass/issues/1562
